src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -366,6 +368,8 @@ public PackInfo(string id, string version, WorkloadPackKind kind, string path)
 
             public WorkloadPackKind Kind { get; }
 
+            public string? AliasedPackageId { get; }
```

Comment:
is this nullable just to aid testing?  I don't think the resolver will ever create a PackInfo if path is null.

Reply:
No, this will be null in case that there isn't an alias associated with the current OS, meaning the package is not required.

Reply:
But in that case wouldn't Path also be null and it isn't nullable.

Reply:
No, the path is constructed from the pack id, not the aliased package id, so the path is always present.

Reply:
My read is that WorkloadResolver will never create a PackInfo when the path is null, as it would be in this case so the property need not be nullable and GivenManagedInstallItIgnoresAliasedPacksWithNoAliasForThisOs() isn't needed.  see https://github.com/dotnet/sdk/pull/16956

Reply:
Right, the path can never null, which is fine because the path is determined based on the pack id, which can also never be null. However, the aliased package id could be null in the case where there is no valid alias for the current OS (which means this pack is not required).

Reply:
> No, the path is constructed from the pack id, not the aliased package id, so the path is always present.

This isn't true after #16956 which is what I wanted to make clear

Reply:
If the pack aliases to nothing I would probably just return a null PackInfo?